### PR TITLE
Break dependency cycles within `:computation_client`

### PR DESCRIPTION
**Pull request by will-cromar**
`XrtComputationClient` and `PjRtComputationClient` both depend on `ComputationClient`, which in turn depends on them for the implementation of `ComputationClient::Create`. This forces us to create one huge build rule that obscures which dependencies are required for each runtime.

- Create separate build rules for `computation_client` (interface only), `pjrt_computation_client`, and `xrt_computation_client`.
- Move all XRT-specific logic in `computation_client` to `xrt_computation_client`
- Break the circular dependency by moving Client creation to a new target `runtime` (I'm open to better names here)
  - Rename `Get()` to `GetClient()`
- Move XRT local service entrypoint out of `ComputationCLient`.

Next steps: Add a build option to exclude XRT from the build. Cutting out XRT from the build reduces the TPU binary size to just 95MB from 141MB!

### CODE REVIEW

**Comment by JackCaoG** [comment]
Yay, this is exciting!

File: test/cpp/test_replication.cpp [hunk 1]
```diff
@@ -70,8 +69,7 @@ void TestSingleReplication(
   mwait.Wait();
 
   for (size_t i = 0; i < results.size(); ++i) {
-    auto literals =
-        xla::ComputationClient::Get()->TransferFromServer(results[i]);
+    auto literals = xla::GetClient()->TransferFromServer(results[i]);
```
[the remark this answers is not in the document]
Comment (will-cromar) [reply test/cpp/test_replication.cpp:72]: We can just make this `GetComputationClient`. Although, should any of `third_party/xla_client` still be in the `xla::` namespace, or should it be in our own namespace?

[the remark this answers is not in the document]
Comment (JackCaoG) [reply test/cpp/test_replication.cpp:72]: I found this `xla` namespace to be confusing.. we no longer move this repo into TF and rebuild, I think we should give it a different namespace. That can happen in a separate pr through.

File: third_party/xla_client/BUILD [hunk 2]
```diff
@@ -56,101 +56,163 @@ cc_test(
...
+    "@org_tensorflow//tensorflow/compiler/xla/client:xla_computation",
+    "@org_tensorflow//tensorflow/compiler/xla/pjrt/distributed",
+    "@org_tensorflow//tensorflow/compiler/xla/pjrt/gpu:se_gpu_pjrt_client",
+    "@org_tensorflow//tensorflow/compiler/xla/pjrt:tpu_client",
+    "@org_tensorflow//tensorflow/compiler/xla/pjrt:pjrt_client",
+    "@org_tensorflow//tensorflow/compiler/xla/pjrt:tfrt_cpu_pjrt_client",
+    "@org_tensorflow//tensorflow/compiler/xla/pjrt:pjrt_c_api_client",
+    "@org_tensorflow//tensorflow/tsl/profiler/lib:traceme",
+    "@com_google_absl//absl/strings",
+    "@com_google_absl//absl/types:span",
```
Comment (JackCaoG) [inline third_party/xla_client/BUILD:127]: wow this is clean!

Comment (JackCaoG) [reply third_party/xla_client/BUILD:127]: let's run TPU CI to verify

File: test/cpp/test_replication.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (stgpetrovic) [reply test/cpp/test_replication.cpp:72]: I would just move `third_party/xla_client` under `torch_xla/cpp/xla` or the like too (in a separate PR).

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Pull request merged**